src/main/check.cpp

```
@@ -184,30 +184,37 @@ int main(int argc, char** argv) {
         }
-        if (!desired_section.empty() && raw_progs.empty()) {
-            // We could not find the desired program, so get the full list
-            // of possibilities.
-            raw_progs = read_elf(filename, string(), string(), ebpf_verifier_options, &platform);
-        }
-        for (const RawProgram& raw_prog : raw_progs) {
-            std::cout << "section=" << raw_prog.section_name << " function=" << raw_prog.function_name << std::endl;
+        for (const ElfProgramInfo& prog : elf.list_programs()) {
+            std::cout << "section=" << prog.section_name << " function=" << prog.function_name;
+            if (prog.invalid) {
+                std::cout << " [invalid: " << prog.invalid_reason << "]";
```

Cppcheck (2.19.0)

[style] The function 'entail_aux' is never used.

(unusedFunction)